lib/puppet/functions/parsepson.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+#
+# parsepson.rb
+#
+# @summary
+#   This function accepts PSON, a Puppet variant of JSON, as a string and converts
+#   it into the correct Puppet structure
+#
+# @example How to parse pson
+#   $data = parsepson("{any valid pson: string}")
+#
+# For more information on PSON please see the following link:
+# https://puppet.com/docs/puppet/7/http_api/pson.html
+#
+Puppet::Functions.create_function(:parsepson) do
+  # @param pson_string A valid PSON string
+  # @param default An optional default to return if parsing the pson_string fails
+  # @return [Data]
+  dispatch :parsepson do
+    param          'String', :pson_string
```

Comment:
Since an empty string is invalid, should this be at least non-empty?
```suggestion
    param          'String[1]', :pson_string
```

Reply:
Updated

Reply:
Have walked myself back on this a bit.
Setting it to a `String[1]` overrides the ability to set a default return value.
Since I'm not entirely sure of all the circumstances that the setting of this default would be valuable I think it's best to leave it as a standard `String` value.